src/app/inventory/store/objectives.ts

```diff
@@ -61,3 +62,53 @@ export function isBooleanObjective(
       (!objectiveDef.allowOvercompletion || !objectiveDef.showValueOnComplete))
   );
 }
+
+export function isTrialsPassage(item: DimItem, defs: D2ManifestDefinitions) {
+  if (item.objectives?.length === 3 && item.isExotic) {
+    for (const objective of item.objectives) {
+      const objectiveDef = defs.Objective.get(objective.objectiveHash);
+      if (isFlawlessObjective(objective, objectiveDef)) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+export function isFlawlessPassage(
+  objectives: DestinyObjectiveProgress[] | null,
+  defs: D2ManifestDefinitions
+) {
+  if (objectives && objectives.length === 3) {
+    for (const objective of objectives) {
+      if (isFlawlessObjective(objective, defs.Objective.get(objective.objectiveHash))) {
+        return objective.complete;
+      }
+    }
+    return false;
```

Comment:
No need for this one

Reply:
This function has been simplified in the new code.